### PR TITLE
Do not render truncated links in markdown

### LINKED ISSUE #31780: Links on dashboard summary are malformed
### Description

When the link in an issue comment is cutoff in the dashboard the link is malformed. 
original link: https://10.1.2.3/
cut off link: [https://10](https://0.0.0.10/).…

Expected result
[https://10](https://10.1.2.3/).…

Original example comment
```
Lorem Ipsum is simply dummy text of the printing and typesetting
Lorem Ipsum is simply dummy text of the printing and typesetting
Lorem Ipsum is simply dummy text of the printing and
https://10.1.2.3
```

### Gitea Version

demo and 1.22.1

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

image

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

gitea demo

### Database

None

### PR DESCRIPTION
**Pull request by wxiaoguang**

### CODE REVIEW

File: modules/util/truncate.go [hunk 1]
```diff
@@ -14,6 +14,10 @@ const (
 	asciiEllipsis = "..."
 )
 
+func IsLikelySplitLeftPart(s string) bool {
```
Comment (delvh) [inline modules/util/truncate.go:17]: `IsLikelyTruncated`?

Comment (wxiaoguang) [reply modules/util/truncate.go:17]: Nope, "Truncated" doesn't have such "..."

It matches the original function below "SplitStringAtByteN"

Comment (wxiaoguang) [reply modules/util/truncate.go:17]: I think we can refactor these "truncate" related function in the future together, while keep this PR simple to backport

**Review by delvh — APPROVED** [review]

**Review by lunny — APPROVED** [review]

File: modules/util/truncate.go (hunk 1 above)
Comment (wxiaoguang) [reply modules/util/truncate.go:17]: ->  Refactor "string truncate" #32984

**Pull request merged**